rj_gameplay/stp/tactic/__init__.py

```diff
@@ -147,8 +147,10 @@ def create_requests(self, num_requests: int, **kwargs) -> List[role.RoleRequest]
         return [self.create_request(**kwargs) for _ in range(num_requests)]
 
     @abstractmethod
-    def tick(self, role_results: RoleResults, props: PropT) -> List[skill.ISkill]:
+    def tick(self, world_state: rc.WorldState, role_results: RoleResults,
+             props: PropT) -> List[skill.ISkill]:
```

Comment:
Just so I understand, how is props different than world_state? And how come none of the implementations of this interface take in props?

Reply:
The infrastructure of our gameplay system was designed by @oswinso and built out by Kyle, Hussain, Yudai and I. This is an example of an issue stemming from that: "props" is short for properties, which I think was intended to be a way for us to pass data from the previous tick of a tactic to the next, rather than recomputing everything every tick from world_state. We currently don't use props (see any tactic's [empty compute_props implementation](https://github.com/RoboJackets/robocup-software/blob/comp2021-merge/rj_gameplay/rj_gameplay/tactic/wall_tactic.py#L114-L115)).

Reply:
It might be worth removing the method entirely from the interface and making a note in GH issues that this is the way we can pass state around. For now, it seems to be overengineering.